### PR TITLE
Bug 1116175 - Write a gaia UI test for moztrap test: Share a music as a ...

### PR DESCRIPTION
**Pull request by mwargers**
...ringtone

### CODE REVIEW

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=714e761489a0)

File: tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        # wait until the player view is shown, then pause the play
+        play_time = time.strptime('00:02', '%M:%S')
+        Wait(self.marionette).until(lambda m: player_view.player_elapsed_time >= play_time)
+
+        activities = player_view.tap_share_button()
+        ringtone = activities.share_to_ringtones()
+
+        # This is to make sure the save button is activated
+        time.sleep(0.5)
```
Comment (npark-mozilla) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py]: would it be possible to use Wait() statement until the save button is visible, then tap it?

Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py]: Ah, just found out that the save button is  disabled at first. So I think I just need to check for when the disable attribute has been removed, then I won't get an intermittent failure there.

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=a5c39c220fd2)

File: tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py (hunk 1 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py]: Ok, that worked, so I removed the time.sleep() call and used that.

Comment (npark-mozilla) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py]: awesome!

File: tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        system = System(self.marionette)
+        self.marionette.switch_to_frame()
+        system.wait_for_notification_toaster_displayed(message="Ringtone set as default.")
+        system.wait_for_notification_toaster_not_displayed()
+
+        settings = Settings(self.marionette)
+        settings.launch()
+        sound = settings.open_sound_settings()
+
+        # desktop b2g doesn't have this option visible, currently
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py]: I would open a bug in Gaia::Settings to track this issue on b2g-desktop.

File: tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        music_app.wait_for_music_tiles_displayed()
+
+        # switch to songs view, and play the first one on the list
+        list_view = music_app.tap_songs_tab()
+        songs = list_view.media
+        self.assertGreater(len(songs), 0, 'The ogg file could not be found')
+        player_view = songs[0].tap_first_song()
+
+        # wait until the player view is shown, then pause the play
+        play_time = time.strptime('00:02', '%M:%S')
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/music/test_music_share_ringtone.py]: Like you said offline, waiting 2 seconds might not be valuable here. We still need to wait until the player_view is displayed though.

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b80b3d39d802)

**Comment by try-server-hook** [comment]
[mwargers Martijn (mwargers)](https://github.com/mwargers) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=922edc8a98fa)

**Pull request merged**